src/main/java/com/spotify/dns/DnsSrvResolvers.java

```diff
@@ -28,18 +40,139 @@
  * Provides utility methods for instantiating and working with DnsSrvResolvers.
  */
 public final class DnsSrvResolvers {
+
   private static final int DEFAULT_DNS_TIMEOUT_SECONDS = 5;
 
   public static DnsSrvResolverBuilder newBuilder() {
-    return new DnsSrvResolverBuilder(null, false, false, SECONDS.toMillis(DEFAULT_DNS_TIMEOUT_SECONDS));
+    return new DnsSrvResolverBuilder();
+  }
+
+  public static DnsSrvWatcherBuilder<LookupResult> newWatcherBuilder(DnsSrvResolver resolver) {
+    checkNotNull(resolver, "resolver");
+
+    return new DnsSrvWatcherBuilder<LookupResult>(resolver, Functions.<LookupResult>identity());
+  }
+
+  public static <T> DnsSrvWatcherBuilder<T> newWatcherBuilder(
+      DnsSrvResolver resolver,
+      Function<LookupResult, T> resultTransformer) {
+
+    checkNotNull(resolver, "resolver");
+    checkNotNull(resultTransformer, "resultTransformer");
+
+    return new DnsSrvWatcherBuilder<T>(resolver, resultTransformer);
+  }
+
+  public static final class DnsSrvWatcherBuilder<T> {
+
+    private final DnsSrvResolver resolver;
+    private final Function<LookupResult, T> resultTransformer;
+
+    private final boolean polling;
+    private final long pollingInterval;
+    private final TimeUnit pollingIntervalUnit;
+
+    private final DnsSrvWatcherFactory<T> dnsSrvWatcherFactory;
+
+    private final ScheduledExecutorService scheduledExecutorService;
+
+    private DnsSrvWatcherBuilder(
+        DnsSrvResolver resolver,
+        Function<LookupResult, T> resultTransformer) {
+      this(resolver, resultTransformer, false, 0, null, null, null);
+    }
+
+    private DnsSrvWatcherBuilder(
+        DnsSrvResolver resolver,
+        Function<LookupResult, T> resultTransformer,
+        boolean polling,
+        long pollingInterval,
+        TimeUnit pollingIntervalUnit,
```

Comment:
The nulls are just internal to the builder. All methods do null checks. I tried to stay away from @Nullable annotations and exposed nullables as much as possible.